pkg/platform/kube/platform.go

```diff
@@ -656,20 +656,37 @@ func (p *Platform) GetDefaultInvokeIPAddresses() ([]string, error) {
 	return []string{}, nil
 }
 
+func (p *Platform) GetScaleToZeroConfiguration() (*platformconfig.ScaleToZero, error) {
+	switch configType := p.Config.(type) {
+	case *platformconfig.Config:
+		return &configType.ScaleToZero, nil
+	case *config.Configuration:
+		return nil, nil
+	default:
+		return nil, errors.New("Not a valid configuration instance")
+	}
+}
+
 func (p *Platform) setScaleToZeroSpec(functionSpec *functionconfig.Spec) error {
+
+	// If function already has scale to zero spec, don't override it
 	if functionSpec.ScaleToZero != nil {
 		return nil
 	}
 
-	switch configType := p.Config.(type) {
-	case *platformconfig.Config:
+	scaleToZeroConfiguration, err := p.GetScaleToZeroConfiguration()
+	if err != nil {
+		return errors.Wrap(err, "Failed getting scale to zero configuration")
+	}
+
+	if scaleToZeroConfiguration == nil {
```

Comment:
because of `case *config.Configuration:` from above? is it a legit scenario?

Reply:
Yup
When deploying using nuctl in a Kubernetes environment, it will be a kube platform but with `case *config.Configuration` which has no scale to zero configuration, in that case we do nothing (in the `setScaleToZeroSpec`)

Reply:
Add a `FIXME` note to the relevant code section with this explanation